### ISSUE TITLE
replace go-sql-connector/mysql with siddontang/go-mysql/client to improve dumpling's performance

### ISSUE THREAD

**Issue by lichunzhu** [pingcap/dumpling#126]
## Feature Request

**Is your feature request related to a problem? Please describe:**

Currently dumpling's performance is only half of mydumper. After tracing the torch graph and doing some simple test we find that dumpling costs a lot of time in fetch one row. That's because `driver.Value` in `database/sql` package is an `interface{}` type variable. When we convert `[]byte` type varible to `interface{}` type, it will use `mallocgc` in `convTslice` to do that which will cost a lot of time, but we can't change `driver.Value`. One solution is to abandon the usage of `database/sql` and directly use the `[]byte` value readed from mysql server. But this is a huge change for dumpling.

**Reference: **
1. convertion code in go-mysql: https://github.com/go-sql-driver/mysql/blob/73dc904a9ece5c074295a77abb7a135797a351bf/packets.go#L770
2. dumpling torch graph:
![image](https://user-images.githubusercontent.com/21104942/89268256-dc5a3c00-d66a-11ea-84e1-5b39d85165ce.png)
3. mydumper torch graph:
![image](https://user-images.githubusercontent.com/21104942/89268309-f005a280-d66a-11ea-8810-1b52d42e25ef.png)
4. code to test the effiency when assign `[]byte` to `interface{}`
https://gist.github.com/lichunzhu/2433d332b4bfc57fb7c1aa3f404b4c58

**Describe the feature you'd like:**

replace go-sql-connector/mysql with siddontang/go-mysql/client to improve dumpling's performance

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Optimization:**

**Comment by lance6716** [comment]
append some information which maybe helpful:

`runtime.mallocgc` in `runtime.convTslice` will use per-goroutine's `mcache` to allocate the space for storing internal interface structure. `mcache` has preserved some fixed-size slot for allocation, when ran out, `mcache` will be dynamically resized from other memory.

**Comment by kennytm** [comment]
@Rustin-Liu shouldn't the bot recognize that this entire repo belongs to SIG-Migrate? 😂

**Comment by Rustin-Liu** [comment]
@kennytm https://github.com/tikv/pd/pull/2981 We can add a config for it.

**Comment by kennytm** [comment]
ok thanks

**Comment by kennytm** [comment]
@sylzd hello, there hasn't been much updates since the pick up, would you need some help?

**Comment by sylzd** [comment]
yes, replace driver may not help,it cost double. we are trying to replace database/sql or something else.

**Comment by lichunzhu** [comment]
> @kennytm yes, replace driver may not help,it cost double. we are trying to replace database/sql or something else.
> 
> 1. our debug flame graph result shows that only 7.44% `convTslice` cost.
>    Command:
> 
> ```
> time ./dumpling -h 10.162.1.1 -u dump_test2 -p* -P 4000 --filetype sql -r 10000 --threads 16 -B sbtest -o dumpling_output
> ```
> 
> FlameGraph:
> ![image](https://user-images.githubusercontent.com/10683568/96566619-61630200-12f8-11eb-9352-ddc915c10824.png)
> 
> 1. when we replace `go-sql-driver` with `go-mysql`, `convTslice` only cost 1.22% and `real time` is double than before.(it cost too much to query)
>    ![image](https://user-images.githubusercontent.com/10683568/96568161-35e11700-12fa-11eb-94b5-626b9d85a6cd.png)
>    ![image](https://user-images.githubusercontent.com/10683568/96569614-f1567b00-12fb-11eb-83a6-6c67489b45fc.png)

Do you replace `go-sql-driver` with `go-mysql/driver`? Or do you `go-sql-driver` and `database/sql` with `go-mysql/client`?
